www/nginx/src/opnsense/scripts/nginx/ngx_autoblock.php

```diff
@@ -259,8 +259,20 @@ function cleanup_work_files($work_files)
     // Reading stored banned IPs from config
     $model = new Nginx();
     $alias_ips = [];
-    foreach ($model->ban->iterateItems() as $entry) {
-        $alias_ips[] = (string)$entry->ip;
+    $ban_ttl = intval((string)$model->general->ban_ttl);
+    $expired = False;
+    foreach ($model->ban->iterateItems() as $id => $entry) {
+        if ($ban_ttl && ($ban_ttl > 0) && ((time() - intval((string)$entry->time)) > 3600*$ban_ttl)) {
```

Comment:
put that check before the foreach loop - it will improve the performance if this has to be checked only once. Also cache the time() value